drivers/include/periph/flashpage.h

```diff
@@ -107,6 +107,27 @@ static inline int flashpage_page(void *addr)
 void flashpage_write(int page, void *data);
 
 /**
+ * @brief   Write the given address with the given data and length
+ *
+ * @note The given address must be aligned to the size of the minimal
+ *       writing block allowed by the MCU. This function doesn't erase
+ *       the block to be written automatically, so be sure the block
+ *       is erased before writing it (using flashpage_write function).
+ *       While writing can be limited to small blocks, erasing often
+ *       is limited to bigger blocks, called sectors or rows in some MCUs.
+ *       Thus, take special care when modifying already written data.
+ *
+ * @param[in] target_addr   address to write
+ * @param[in] data          data to write to the address, MUST be aligned
+ *                          to the minimal writable block defined by the
+ *                          MCU.
+ * @param[in] len           length of the data to be written. It MUST not
+ *                          be less than the minimal allowed block write
+ *                          length.
+ */
+void flashpage_write_raw(void *target_addr, void *data, size_t len);
```

Comment:
We'll need variables  for flash alignment and flash block size documented somewhere in this file, so user code has a chance on not tripping over the asserts.

Maybe extern const variables would be an option? Defines cannot documented here without assigning a value, right?

Reply:
Eventually we could add an aligned buffer of a minimal writing size, but that would increase the implementation's size. It's some bytes (from 16 to 64 according to the MCU) but it would allow to always read and write on aligned addresses and lengths.

Reply:
I agree, we can define them in periph_cpu.h and document them there. The doc here is not enough?

Reply:
No. The specific defines need documentation.
Call them something like ```FLASHPAGE_RAW_ALIGNMENT``` and ```FLASHPAGE_RAW_BLOCKSIZE```. The former would be 4 for sam0, the latter 64.